### ISSUE TITLE
Borg 2 mount of a series only mounts a single archive

### ISSUE THREAD

**Issue by witten** [borgbackup/borg#8490]
## Have you checked borgbackup docs, FAQ, and open GitHub issues?

Yes.

## Is this a BUG / ISSUE report or a QUESTION?

Yes.

#### Your borg version (borg -V).

Client: Borg 2.0.0b12 (not using a server for this ticket)

#### Operating system (distribution) and version.

Manjaro stable 24.1.1

#### Hardware / network configuration, and filesystems used.

Lenovo X1 Carbon, no network, ext4

#### How much data is handled by borg?

On the order of KB.

#### Full borg commandline that lead to the problem (leave away excludes and passwords)

```bash
$ /root/borg2.0.0b12 repo-create --encryption repokey-aes-ocb --debug --repo test2.borg
$ /root/borg2.0.0b12 create --debug --repo test2.borg seriesname .
$ /root/borg2.0.0b12 create --debug --repo test2.borg seriesname .
$ mkdir tmp
$ /root/borg2.0.0b12 mount --debug -a seriesname --repo test2.borg tmp
```

## Describe the problem you're observing.

The mount actually succeeds even though a series name matching multiple archives was given. The resulting mount point contains the contents of only one of the archives... maybe?

```bash
$ ls -lR tmp
tmp:
total 0
drwxr-xr-x 1 root root 0 Oct 23 22:05 seriesname

tmp/seriesname:
total 42
drwx------ 1 root root     0 Oct 23 11:37 1.2.borg
drwxr-xr-x 1 root root     0 Oct 23 21:51 t
drwx------ 1 root root     0 Oct 23 13:24 test.borg
-rw------- 1 root root 42861 Oct 23 21:50 test.yaml

tmp/seriesname/1.2.borg:
total 44
-rw------- 1 root root   700 Oct 23 11:37 config
drwx------ 1 root root     0 Oct 23 11:37 data
-rw------- 1 root root   140 Oct 23 11:37 hints.9
-rw------- 1 root root 41258 Oct 23 11:37 index.9
-rw------- 1 root root   190 Oct 23 11:37 integrity.9
-rw------- 1 root root    16 Oct 23 11:37 nonce
-rw------- 1 root root    73 Oct 23 11:37 README

tmp/seriesname/1.2.borg/data:
total 0
drwx------ 1 root root 0 Oct 23 11:37 0
...
```

Instead of this behavior, I'd expect an error like:

```
Command Error: ['seriesname'] needed to match precisely one archive, but matched 2.
```

Alternatively, I'd expect all archives in the series to be mounted in separate subdirectories of the mount point, much like when a whole repository is mounted.

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.

Yes.

#### Include any warning/errors/backtraces from the system logs

Here's the result of the `mount` command:

```
using builtin fallback logging configuration
19 self tests completed in 0.07 seconds
LOCK-INIT: initialising. store: <Store(url='file://test2.borg', levels=[('archives/', [0]), ('config/', [0]), ('cache/', [0]), ('locks/', [0]), ('data/', [2]), ('keys/', [0])])>, stale: 1800s, refresh: 900s.
LOCK-ACQUIRE: trying to acquire a lock. exclusive: False.
LOCK-CREATE: creating lock in store. key: 81d75c9de6d47a88, lock: {'exclusive': False, 'hostid': 'localhost.localdomain@220295161013493', 'processid': 785141, 'threadid': 0, 'time': '2024-10-24T14:43:09.754+00:00'}.
LOCK-ACQUIRE: success! no exclusive locks detected.
security: read previous location 'test2.borg'
security: read manifest timestamp '2024-10-24T05:05:21.868828+00:00'
security: determined newest manifest timestamp as 2024-10-24T05:05:21.868828+00:00
security: repository checks ok, allowing access
mount data cache capacity: 4 chunks
Mounting filesystem
Initializing llfuse
Calling fuse_mount
Calling fuse_lowlevel_new
Calling fuse_session_add_chan
Daemonizing: Foreground process (localhost.localdomain@220295161013493, 785141, 0) is waiting for background process...
Daemonizing: Background process (localhost.localdomain@220295161013493, 785195, 0) is starting...
fuse: mount local repo, going to background: migrating lock.
LOCK-MIGRATE: ('localhost.localdomain@220295161013493', 785141, 0) -> ('localhost.localdomain@220295161013493', 785195, 0).
LOCK-CREATE: creating lock in store. key: 1bff6af50cb7b8cb, lock: {'exclusive': False, 'hostid': 'localhost.localdomain@220295161013493', 'processid': 785195, 'threadid': 0, 'time': '2024-10-24T14:43:09.847+00:00'}.
LOCK-DELETE: deleting lock from store. key: 81d75c9de6d47a88.
Daemonizing: Background process (localhost.localdomain@220295161013493, 785195, 0) has started.
Daemonizing: Foreground process (localhost.localdomain@220295161013493, 785141, 0) is now dying.
```

**Comment by ThomasWaldmann** [comment]
"The mount actually succeeds even though a series name matching multiple archives was given."

I don't see a series name given there.

**Comment by ThomasWaldmann** [comment]
Also please note that in borg2, there is no `NAME` argument. This was left away in favour of `-a`:

```
usage: borg mount [-r REPO] [-a PATTERN] MOUNTPOINT [PATH ...]

Mount archive or an entire repository as a FUSE filesystem

positional arguments:
  MOUNTPOINT            where to mount filesystem
  PATH                  paths to extract; patterns are supported
```

I shortened it a bit to make it better readable.

So, if one wants just one or some archives, one would use `-a` here:

- no `-a` given: mount the whole repo
- `-a name` to only mount archive(s) exactly matching "name"
- `-a sh:name-2024-*` to only mount matching archive(s)
- `-a aid:deadb33f` to mount a specific archive of a series

**Comment by ThomasWaldmann** [comment]
The reason why it was left away:
- borg can mount whole repo or archive(s)
- thus the archive NAME positional argument would need to be optional
- there are also optional PATH arguments
- so there is an ambiguity what the argument right to MOUNTPOINT means, is it a NAME or is it a PATH?

**Comment by witten** [comment]
My bad. I think I pasted the wrong mount command. Here's the one I meant to include:

```
$ /root/borg2.0.0b12 mount --debug -a seriesname --repo test2.borg tmp
```

(I've edited the ticket description accordingly.)

Thanks for the explanation about switching from a positional argument to `-a` for the archive. I guess my question is what's the expected behavior when a series name is given for `-a`? Because right now it just seems to be mounting a single archive. Is that fixed in master by #8461?

**Comment by ThomasWaldmann** [comment]
That command should work with master branch code. Yes, fix is #8461.

The old code can't work as it always ended up with the same directory name as it just used the archive name for that. The fixed code appends the archive id if there are collisions.

**Comment by witten** [comment]
Understood, thanks!
